workspace/data-proxy-sdk/src/data-proxy.ts

```diff
@@ -155,6 +160,61 @@ export class DataProxy {
 		);
 	}
 
+	/**
+	 * Verifies a SEDA Fast proof
+	 * Also checks if the block height is not too old (So that proofs expire)
+	 * @param proof
+	 * @param blockHeight
+	 * @returns
+	 */
+	async verifyFastProof(
+		proof: string,
+	): Promise<
+		Result<
+			{ isValid: boolean; status: string; currentUnixTimestamp: bigint },
+			string
+		>
+	> {
+		const decodedProof = this.decodeSedaFastProof(proof);
+		if (decodedProof.isErr) {
+			return Result.err(
+				`Failed to decode SEDA Fast proof: ${decodedProof.error}`,
+			);
+		}
+
+		const now = BigInt(Date.now());
+		const delta = now - decodedProof.value.unixTimestamp;
+
+		if (delta > this.options.fastMaxProofAgeMs) {
+			return Result.ok({
+				isValid: false,
+				status: "unix_timestamp_too_old",
+				currentUnixTimestamp: now,
+			});
+		}
+
+		// Check if the client is allowed, empty lists means all clients are allowed (debugging only)
```

Comment:
Is this not potentially confusing for operators?

Reply:
Yes, i replaced it with just an error that user must provide a client public key when sedaFast is enabled